controllers/release/release_adapter.go

```diff
@@ -185,14 +185,19 @@ func (a *Adapter) EnsureTargetContextIsSet() (results.OperationResult, error) {
 // will be extracted from the given ReleaseStrategy. The Release's ApplicationSnapshot will also be passed to the
 // release PipelineRun.
 func (a *Adapter) createReleasePipelineRun(releaseStrategy *v1alpha1.ReleaseStrategy, applicationSnapshot *appstudioshared.ApplicationSnapshot) (*v1beta1.PipelineRun, error) {
+	releasePlan, err := a.getReleasePlan()
+	if err != nil {
+		return nil, err
+	}
 	pipelineRun := tekton.NewReleasePipelineRun("release-pipelinerun", releaseStrategy.Namespace).
 		WithOwner(a.release).
 		WithReleaseLabels(a.release.Name, a.release.Namespace, a.release.GetAnnotations()[logicalcluster.AnnotationKey]).
+		WithApplicationLabel(releasePlan.Spec.Application).
```

Comment:
Wondering if we should rename `WithReleaseLabels` to `WithHabcsLabels` and pass here the release and the ApplicationSnapshot to generate all the labels in there. I think it would be cleaner and we would keep a single function to update labels. We wouldn't need a loop anymore as the code would be almost the same as we have right now but adding the new key/value. What do you think @johnbieren?

Reply:
Why not just `WithLabels`?

Reply:
If we are never going to have a release without an application snapshot, i am fine with that approach. I do not like the name `WithHacbsLabels`. I think `WithLabels` is a much better name

Reply:
Also, wouldn't be better to instead of having the strict labeling we call the function as many times we want with the labels we need? Thinking in case we need to add more labels we wouldn't need to touch the function again.

Reply:
I'm ok with `WithLabels`, although is not as descriptive to me. In a function with this name i would expect labels to be passed. What about `WithReleaseAndApplicationLabels`?

Reply:
I don't think we are going to need more, but yeah, could be the case. Thing is that we would have to generate and pass labels from every place we create a PR and every single PR should be a very specific set of labels, so it makes sense IMHO to define them in a single function.

Reply:
`WithReleaseAndApplicationLabels` is fine by me. I just don't think `hacbs` is a good string to add in as we don't have it anywhere else in our code and means nothing without more context

Reply:
change to `WithReleaseAndApplicationLabels`